backend/commons/src/main/java/org/sonarsource/sonarlint/core/commons/NewCodeDefinition.java

```diff
@@ -116,6 +118,11 @@ public boolean isSupported() {
       return true;
     }
 
+    @Override
+    public long getThresholdDate() {
```

Comment:
Shouldn't this method also be implemented in `NewCodeDefinitionWithDate`? Maybe it's the case.
EDIT: I see this abstract class is not used at all, maybe we could remove it?

Reply:
This method is implemented in `NewCodeDefinitionWithDate`. Not sure what you mean exactly.
Also abstract class is holding the common state - `thresholdDate` for children: `NewCodeNumberOfDaysWithDate`, `NewCodePreviousVersion`, `NewCodeSpecificAnalysis`.
By removing the class we will have to re-define the field many times.

If you see how to improve the design, let's discuss or make a PR.
For now I merge as is.